packages/next/client/page-loader.js

```diff
@@ -70,14 +69,54 @@ export default class PageLoader {
         return
       }
 
-      // Load the script if not asked to load yet.
       if (!this.loadingRoutes[route]) {
-        this.loadScript(route)
+        // Make sure we don't load a dependency that's already loaded
+        const loadedModules = new Set(
```

Comment:
My original reason for the sort of complicated version here is that it's only doing one querySelector, as opposed to one per dependency, which depending on the limits we end up going with could be twenty or more files. Is that a valid concern? Or is the query selector fast enough that doing 20 of them won't have any impact?

Reply:
It's pretty well cached internally, though I can't honestly say which would be faster. Normally I'd suggest going with whatever has smaller output size since this is browser code, but I'm not sure there's an easy way to check. I wonder if avoiding `Set` has any value? IIRC it's used elsewhere in the codebase already.

Reply:
Set is a React dependency, but it may provide value when coupled with Preact.